corehq/apps/userreports/reports/view.py

```diff
@@ -703,6 +793,28 @@ def report_preview_data(cls, domain, report_config):
             raise BadBuilderConfigError(DATA_SOURCE_NOT_FOUND_ERROR_MESSAGE)
 
 
+def get_request_data(request):
+    """
+    Returns both URL params and POST data.
+
+    e.g. Given the following incoming POST request ::
+
+        requests.post('http://example.com/?foo=bar', data={'foo': 'baz'})
+
+    the Django request object ``req`` would produce ::
+
+        >>> dict(get_request_data(req))
+        {'foo': ['bar', 'baz']}
```

Comment:
nit: so this would `get_consolidated_request_data`? Can't think of a better name.

looks like you essentially need to cover both get and post request for bs5?
This looks like a very unique case where a param could be present both places.
I'd suggest just keeping it to that instead of making it generic.

```
def _is_bs5_request():
    return 'bs5' in request.GET or 'bs5' in request.POST
```

Reply:
> `get_consolidated_request_data`? 

Fun fact: Early versions of Django used to have this built in as [request.REQUEST](https://docs.djangoproject.com/en/1.8/ref/request-response/#django.http.HttpRequest.REQUEST). So ... `get_request_request()`? :thinking: :smirk: :joy: Just kidding.

I'm planning to remove this. See [this TODO](https://github.com/dimagi/commcare-hq/blob/2aee63ef99e1dcb0ec9e04eaf5941231a1c71814/corehq/apps/campaign/static/campaign/js/dashboard.js#L297-L299) and [this TODO](https://github.com/dimagi/commcare-hq/blob/b54c2797f5cbda8d52366e6e6f6a3e4c18f53feb/corehq/apps/userreports/reports/view.py#L510-L516).

Reply:
ed9c60f48e821a54921fa89eb1cfb8a9b6cc43d6